cmd/signature-v4-utils.go

```diff
@@ -260,3 +262,46 @@ func signV4TrimAll(input string) string {
 	// unicode.IsSpace() internally here) to one space and return
 	return strings.Join(strings.Fields(input), " ")
 }
+
+// checkMetaHeaders will check if the metadata from header/url is equal with the one from signed headers
+func checkMetaHeaders(signedHeadersMap http.Header, r *http.Request) APIErrorCode {
+	metaHeaders := make(http.Header)
+	regex := regexp.MustCompile("^[xX]-[aA]mz-[mM]eta-.*")
+
+	for k := range signedHeadersMap {
+		if regex.MatchString(k) {
+			metaHeaders.Add(k, r.Header.Get(k))
+		}
+	}
+
+	// take values from http header
+	for k := range r.Header {
+		if regex.MatchString(k) {
+			metaHeaders.Add(k, r.Header.Get(k))
+		}
+	}
+
+	// take values from url, if no http header
+	if len(metaHeaders) == 0 {
+		for k := range r.Form {
+			if regex.MatchString(k) {
+				metaHeaders.Add(strings.Title(k), r.Form.Get(k))
+			}
+		}
+	}
+
+	// in case there is no metadata from header or url
+	if len(metaHeaders) == 0 {
+		return ErrNone
+	}
+
+	if _, isKeyExists := metaHeaders["Host"]; !isKeyExists {
```

Comment:
This seems pointless, since you copy all values from `signedHeadersMap` above anyway.

Reply:
About 

> This seems pointless, since you copy all values from signedHeadersMap above anyway.

I added this 
`for k := range signedHeadersMap {
		if regex.MatchString(k) {
			metaHeaders.Add(k, r.Header.Get(k))
		}
	}` by mistake, it should not be there. 
sorry for confusion.

I just take the values from http header or url values, so I need to add the "Host" key so that I can compare the metadata received with the one which was used to signed the url.

Reply:
@ruspaul013 Please update. See if you can get rid of the map allocation without too much complexity.